### PR TITLE
Cluster-autoscaler: basic metrics for ca

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @piosz @jszczepkowski

### CODE REVIEW

File: cluster-autoscaler/cluster_autoscaler.go [hunk 1]
```diff
@@ -50,11 +53,52 @@ var (
 		"How often scale down possiblity is check")
 )
 
+var (
```
Comment (wojtek-t) [inline cluster-autoscaler/cluster_autoscaler.go]: Can you please move metrics to a separate file (or even package)?
As an example:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apiserver/metrics/metrics.go

File: cluster-autoscaler/cluster_autoscaler.go [hunk 2]
```diff
@@ -50,11 +53,52 @@ var (
...
+	lastTimestamp = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "cluster_autoscaler",
+			Subsystem: "main",
```
Comment (wojtek-t) [inline cluster-autoscaler/cluster_autoscaler.go]: I think we don't set subsystem anywhere else. But that's probably not a big problem.

File: cluster-autoscaler/cluster_autoscaler.go [hunk 3]
```diff
@@ -169,7 +216,15 @@ func main() {
 				if len(unschedulablePodsToHelp) == 0 {
 					glog.V(1).Info("No unschedulable pods")
 				} else {
+					scaleUpStart := time.Now()
+					lastTimestamp.WithLabelValues("scale_up").Set(float64(scaleUpStart.Unix()))
 					scaledUp, err := ScaleUp(unschedulablePodsToHelp, nodes, migConfigs, gceManager, kubeClient, predicateChecker, recorder)
+
+					duration.WithLabelValues("scale_up").Observe(
```
Comment (wojtek-t) [inline cluster-autoscaler/cluster_autoscaler.go]: Can you opaque these 4 lines into function? The same lines (that different only with the string and start time) you have in 4 different places

Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done.

File: cluster-autoscaler/cluster_autoscaler.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done.

File: cluster-autoscaler/cluster_autoscaler.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done.

File: cluster-autoscaler/cluster_autoscaler.go [hunk 4]
```diff
@@ -215,7 +244,13 @@ func main() {
...
 					}
 				}
+				updateDuration("main", loopStart)
 			}
 		}
 	}
 }
+
+func updateDuration(label string, start time.Time) {
+	duration.WithLabelValues("main").Observe(durationToMicro(start))
```
Comment (wojtek-t) [inline cluster-autoscaler/cluster_autoscaler.go]: s/main/label/

same below

File: cluster-autoscaler/cluster_autoscaler.go [hunk 5]
```diff
@@ -55,6 +58,12 @@ func main() {
 		"Can be used multiple times. Format: <min>:<max>:<migurl>")
 	flag.Parse()
 
+	http.Handle("/metrics", prometheus.Handler())
```
Comment (wojtek-t) [inline cluster-autoscaler/cluster_autoscaler.go]: git add on the new file please

File: cluster-autoscaler/cluster_autoscaler.go (hunk 4 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done.

File: cluster-autoscaler/cluster_autoscaler.go (hunk 5 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done.

File: cluster-autoscaler/cluster_autoscaler.go [hunk 6]
```diff
@@ -109,6 +118,9 @@ func main() {
 		select {
 		case <-time.After(time.Minute):
 			{
+				loopStart := time.Now()
+				lastTimestamp.WithLabelValues("main").Set(float64(loopStart.Unix()))
```
Comment (wojtek-t) [inline cluster-autoscaler/cluster_autoscaler.go]: how about making this also a function, like:
updateLastTimestamp(labels, time)

**Comment by wojtek-t** [comment]
One comment - other than that lgtm

File: cluster-autoscaler/cluster_autoscaler.go (hunk 6 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done

**Pull request merged**